jwst/outlier_detection/utils.py

```diff
@@ -30,8 +31,104 @@ def create_cube_median(cube_model, maskpt):
     return median
 
 
+def create_median_library(resampled_models, maskpt, on_disk=True, buffer_size=10.0):
+    """Create a median image from the singly resampled images.
+    resampled_models is expected to be a ModelLibrary for imaging modes.
+    """
+    # Compute the weight threshold for each input model
+    weight_thresholds = []
+    with resampled_models:
+        for resampled in resampled_models:
+            weight = resampled.wht
+            weight_threshold = compute_weight_threshold(weight, maskpt)
```

Comment:
```suggestion
            weight_threshold = compute_weight_threshold(resampled.wht, maskpt)
```
Does this work? It allows `resampled.wht` to fall out of scope with `resampled`.

Reply:
I'm changing it, but why wouldn't it work?

Reply:
Thanks! I wasn't sure if `weight` was also used below.